### PR TITLE
Pass custom scrollToHandler through if defined in options.

### PR DESCRIPTION
**Pull request by sirvine**
This exposes the scrollToHandler option so that a custom function can be passed in. For example:

``` javascript
    let topScrollHandler = () => {
      Ember.$('#main-scroll-container').animate({
        scrollTop: 0
      }, 600);
    };
    
    let steps = [
      {id: 'step-one',
       options: {
          attachTo: '#example-element right',
          title: 'Example Step Title',
          text: [stepOneLabel],
          scrollTo: true,
          scrollToHandler: topScrollHandler
        }
      }
    ];

```

### CODE REVIEW

File: addon/services/tour.js [hunk 1]
```diff
@@ -448,18 +448,20 @@ export default Service.extend(Evented, {
...
-      // Allow scrollbar scrolling so scrollTo works.
-      currentStep.options.scrollToHandler = (elem) => {
-        $window.disablescroll({
-          handleScrollbar: false
-        });
-
-        if (typeof elem !== 'undefined') {
-          elem.scrollIntoView();
-        }
+      if (currentStep.options.scrollToHandler === 'undefined') {
```
Comment (rwwagner90) [inline addon/services/tour.js]: Wouldn't this need to be typeof? Couldn't you just say `if (!currentStep.options.scrollToHandler)`

**Comment by rwwagner90** [comment]
Tests seem to be failing, not sure why 😞

**Comment by sirvine** [comment]
I guess that was it. Anyway, I wanted to expose this option because it was useful to me for fine-tuning the scroll position and animation. Your original default is probably the most widely-compatible approach anyway. Thanks!

**Comment by rwwagner90** [comment]
@sirvine sorry to be picky, but could you write a test for the default as well, to ensure everything still scrolls as it should, when a custom one is not passed?

Also, could you document this new option in the README please?

**Comment by sirvine** [comment]
No problem. Let me know if you need any adjustments. I struggled a bit with the best way to test for proper operation in the absence of the custom `scrollToHandler` option. In the end, my test for that condition is largely redundant with the existing `scrollTo works with disableScroll on` test.

**Comment by rwwagner90** [comment]
Thanks for making those changes @sirvine! It all looks good, from a code perspective, but for some reason, the tests are failing on Travis for Ember 1.13 😢

**Comment by rwwagner90** [comment]
Looks like it was a fluke that the tests failed. Ran again and they passed. Going to go ahead and merge. Thanks! Any other features you'd like to see implemented @sirvine?

**Comment by sirvine** [comment]
I noticed the same (or similar problem) with Travis on one of my earlier commits, for the record. I have only begun to play with shepherd in my app, so I may have more to come. For now, it is quite useful to me. Funny enough, your question prompted to me to look at the shepherd.js docs, and I see that scrollToHandler is undocumented since 2014! So now maybe we can say that ember-shepherd has better documentation of this feature than the library that it wraps! 

For some context, the reason that I felt it would be useful to add custom scrolling is that I worried that it would be too jarring for my users if the first step scrolled down to the element top. At first, I just turned `scrollTo` to false on the first step, but that caused problems when backing up from step 2 to step 1. As a result, I wanted a scrollTo that pinned step 1 at the top of the window. 

Thank you!

**Pull request merged**